src/Network/DNS.hs

```diff
@@ -161,7 +165,13 @@ queryRaw (Class cls) (Name name) qtype = withCResState $ \stptr -> do
                 unless (errno == eOK) $
                     throwErrno "res_query"
 
-                fail "res_query(3) failed"
+                h_errno <- c_get_h_errno
+                case h_errno of
+                  1 -> throw DnsHostNotFound
+                  2 -> throw DnsNoData
+                  3 -> throw DnsNoRecovery
+                  4 -> throw DnsTryAgain
+                  _ -> fail "res_query (3) failed"
```

Comment:
Similar messages (`res_send(3) failed` and `res_mkquery(3) failed` have no space between the function name and the man section tag `(3)`, consider removing the space here for consistency.